core/src/main/python/validate.py

```diff
@@ -110,6 +111,12 @@ def __perform_model_file_validation(model_file_name, model_context):
         variable_map = model_validator.load_variables(model_context.get_variable_file())
         model_dictionary = cla_helper.merge_model_files(model_file_name, variable_map)
 
+        # apply filters to merged model
+        if variable_map:
+            # if there are any variables than substitute them in the model
+            variables.substitute(model_dictionary, variable_map, model_context)
```

Comment:
the model should still have variables unresolved before going into `validate_in_standalone_mode`, so that any missing variables can be logged there.

Reply:
The change now invokes filtering after validating the merged models (i.e calling **validate_in_standalone_mode**).  We will validate a second time (and re-persist) if filters were applied.